dbt/adapters/duckdb/plugins/gsheet.py

```diff
@@ -0,0 +1,54 @@
+from dataclasses import dataclass
+from typing import Dict
+from typing import Literal
+
+import gspread
+import pandas as pd
+
+from . import Plugin
+from . import PluginConfig
+from ..utils import SourceConfig
+
+
+@dataclass
+class GSheetConfig(PluginConfig):
+    method: Literal["service", "oauth"]
+
+    def client(self):
+        if self.method == "service":
+            return gspread.service_account()
+        else:
+            return gspread.oauth()
+
+
+class GSheetPlugin(Plugin):
+    def __init__(self, config: Dict):
+        self._config = GSheetConfig.from_dict(config)
+        self._gc = self._config.client()
+
+    def load(self, source_config: SourceConfig):
+        doc = None
+        if "title" in source_config.meta:
+            doc = self._gc.open(source_config.meta["title"])
+        elif "key" in source_config.meta:
+            doc = self._gc.open_by_key(source_config.meta["key"])
+        elif "url" in source_config.meta:
+            doc = self._gc.open_by_url(source_config.meta["url"])
+        else:
+            raise Exception("Source config did not indicate a method to open a GSheet to read")
+
+        sheet = None
+        if "worksheeet" in source_config.meta:
```

Comment:
maybe triple `eee` is a typo?

Reply:
yay, thank you! What I get for not testing this plugin out yet ;-)